.github/workflows/release.yml

```diff
@@ -1,52 +1,205 @@
+          
+          // Set the tag (which has the leading "v") prefix.
+          const tag = `v${version}`;
+          core.info(`👾 Computed tag as: "${tag}"`)
+          core.setOutput('tag', tag)
+          
+          // Verify the commit is signed.
+          if (!commitResult.data.commit.verification.verified) {
+            core.setFailed(`❌ Commit is not signed`)
+            return;
+          }
+          
+          // Verify the email matches the expected committer.
+          const expectedEmail = process.env.EXPECTED_EMAIL;
+          const gotEmail = commitResult.data.commit.author.email;
+          if (gotEmail !== expectedEmail) {
+            core.setFailed(`❌ Commit author is "${gotEmail}", expected "${expectedEmail}"`);
+            return;
+          }
+          
+          // Compute prerelase.
+          const prerelease = ['-', 'pre', 'alpha', 'beta', 'preview'].some((v) => version.includes(v));
+          
+          // Create the release.
+          const response = await github.rest.repos.createRelease({
+            owner: context.repo.owner,
+            repo: context.repo.repo,
+            tag_name: tag,
+            target_commitish: headCommit.id,
+            name: tag,
+            generate_release_notes: true,
+            prerelease: prerelease,
+            draft: true,
+            make_latest: 'legacy',
+          });
+          core.setOutput('created', true);
+          core.info(`✅ Created release "${response.data.name}" at ${response.data.html_url}`);
+  build:
+    uses: './.github/workflows/build.yml'
+    needs:
+    - 'create-release'
+    with:
+      # version is the same as `${UNIQUE_ID}`, but there's no way to reference
+      # the "env" context from above in reusable workflows[1].
+      #
+      # [1]: https://github.com/orgs/community/discussions/26671
+      version: '${{ needs.create-release.outputs.version }}'
 
-    - name: 'Run GoReleaser'
-      uses: 'goreleaser/goreleaser-action@7ec5c2b0c6cdda6e8bbb49444bc797dd33d74dd8' # ratchet:goreleaser/goreleaser-action@v5
+  publish-release:
+    runs-on: 'ubuntu-latest'
+    permissions:
+      contents: 'read'
+      id-token: 'write'
+    needs:
+    - 'create-release'
+    steps:
+    - name: 'Mint token'
+      id: 'mint-token'
+      uses: 'abcxyz/github-token-minter/.github/actions/mint-token@main' # ratchet:exclude
       with:
-        distribution: 'goreleaser'
-        version: 'v1.18.2'
-        args: 'release --clean'
+        wif_provider: '${{ vars.TOKEN_MINTER_WIF_PROVIDER }}'
+        wif_service_account: '${{ vars.TOKEN_MINTER_WIF_SERVICE_ACCOUNT }}'
+        service_audience: '${{ vars.TOKEN_MINTER_SERVICE_AUDIENCE }}'
+        service_url: '${{ vars.TOKEN_MINTER_SERVICE_URL }}'
+        requested_permissions: |-
+          {
+            "repositories": ["${{ github.event.repository.name }}"],
+            "permissions": {
+              "contents": "write"
+            }
+          }
+
+    - name: 'Publish release'
```

Comment:
Are we not publishing binaries?

Reply:
No, this is a library.

Future versions might depending on where we decide to put the server part.